rpcs3/CMakeLists.txt

```diff
@@ -166,6 +166,26 @@ find_package(OpenGL REQUIRED)
 find_package(OpenAL REQUIRED)
 if (NOT WITHOUT_LLVM)
 	find_package(LLVM 4.0 CONFIG)
+  if (NOT LLVM_FOUND)
+    message("System LLVM was not found, LLVM will be built from the submodule.")
+
+    set(LLVM_TARGETS_TO_BUILD "X86" CACHE INTERNAL "")
+    option(LLVM_BUILD_RUNTIME OFF)
+    option(LLVM_BUILD_TOOLS OFF)
+    option(LLVM_INCLUDE_DOCS OFF)
+    option(LLVM_INCLUDE_EXAMPLES OFF)
+    option(LLVM_INCLUDE_TESTS OFF)
+    option(LLVM_INCLUDE_TOOLS OFF)
+    option(LLVM_INCLUDE_UTILS OFF)
+    option(WITH_POLLY OFF)
+
+    # LLVM needs to be built out-of-tree
+    add_subdirectory(../llvm ../llvm_build)
+    set(LLVM_DIR "${CMAKE_CURRENT_BINARY_DIR}/../llvm_build/lib/cmake/llvm/")
```

Comment:
This would break the out-of-tree build (which is required now), because `CMAKE_SOURCE_DIR` points to the RPCS3 source tree, not the build output. The path `llvm_build/lib/cmake/llvm/` only exists after cmake is first run.

If I put `CMAKE_BINARY_DIR` instead that'd fall more or less  into the the same pitfall LLVM's CMake put us on: you can't properly add a LLVM subproject without triggering LLVM own protection against in-source builds. If RPCS3 for some reason or another is added as a sub project `CMAKE_BINARY_DIR` will point to the parent project. So I prefer `CMAKE_CURRENT_BINARY_DIR` here.

Reply:
Alright, makes sense to me

Reply:
Does this allow `LLVM_DIR` to be set by the user? I don't think so, it would be nice.

Reply:
Yes, it does allow LLVM_DIR to be set be the user @ranisalt . This is because it first checks for llvm, to do that it can check using `LLVM_DIR`, if it can't find LLVM by looking into the default locations, and by using the user's set LLVM_DIR, then this PR makes it so that it forces where to find LLVM.

Reply:
Very good!